FirebaseAuthUI/FUIPrivacyAndTermsOfServiceView.m

```diff
@@ -0,0 +1,118 @@
+//
+//  Copyright (c) 2016 Google Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#import "FUIPrivacyAndTermsOfServiceView.h"
+
+#import <FirebaseAuth/FirebaseAuth.h>
+#import "FUIAuth.h"
+#import "FUIAuthStrings.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation FUIPrivacyAndTermsOfServiceView
+
+#pragma mark - Public
+
+- (void)useFullMessage {
+  NSAttributedString *fullMessage = [self fullPrivacyPolicyAndTOSMessage];
+  self.attributedText = fullMessage;
+  self.textAlignment = NSTextAlignmentLeft;
+}
+
+- (void)useFooterMessage {
+  NSAttributedString *footerMessage = [self footerPrivacyPolicyAndTOSMessage];
+  self.attributedText = footerMessage;
+  self.textAlignment = NSTextAlignmentRight;
+}
+
+#pragma mark - Private
+
+- (NSAttributedString *)fullPrivacyPolicyAndTOSMessage {
+  if (![FUIAuth defaultAuthUI].TOSURL.absoluteString.length &&
```

Comment:
It's not a good idea to depend on the default authUI instance, since it may not exist (see #371). However, most people commonly just use the default auth, so we should make it easy to use by default. Can you change this to inject the authUI instance (either method- or class-level) instead of just referencing the default instance?

Reply:
Done.